File: Java Script - Introduction to Programming/9.9. Functions/13. Notifications.js
function notifications([...arg]) {
    arg.shift()
    processMessage(arg)

    function processMessage(info) {
        while (info.length !== 0) {

            let operation = info.shift()

            if (operation === 'success') {
                console.log(showSuccessMessage(info.shift(), info.shift()))
            } else if (operation === 'warning') {
                console.log(showWarningMessage(info.shift()))
            } else { //error
                console.log(showErrorMessage(info.shift(), info.shift(), info.shift()))
            }
        }
    }

    function showSuccessMessage(operation, message) {
        return `Successfully executed ${operation}.\n${'='.repeat(operation.length + 23)}\n${message}.\n`
    }

    function showWarningMessage(message) {
        return `Warning: ${message}.\n${'='.repeat(message.length + 10)}\n`
    }

    function showErrorMessage(operation, message, errorCode) {
        return `Error: Failed to execute ${operation}.\n${'='.repeat(operation.length +26)}\nReason: ${message}.\nError code: ${errorCode}.\n`
    }
}


// notifications([
//     '4',
//     'error',
//     'credit card purchase',
//     'Invalid customer address',
//     '500',
//     'warning',
//     'Email not confirmed',
//     'success',
//     'user registration',
//     'User registered successfully',
//     'warning',
//     'Customer has not email assigned',

// ])

File: Pyrhon - Introduction to Programming/9.9. Functions/13. Notifications.py
def notifications(message_count):

    def show_success_message(operation, message):
        return f'Successfully executed {operation}.\n{"=" * (len(operation) + 23)}\n{message}.\n'


    def show_warning_message(message):
        return f'Warning: {message}.\n{"=" * (len(message) + 10)}\n'


    def show_error_message(operation, message, error_code):
        return f'Error: Failed to execute {operation}.\n{"=" * (len(operation) + 26)}\nReason: {message}.\nError code: {error_code}.\n'


    def process_message(message_count):

        while message_count:
            message_count -= 1
            operation = input()

            if operation == 'success':
                print(show_success_message(input(), input()))
            elif operation == 'warning':
                print(show_warning_message(input()))
            else:
                print(show_error_message(input(), input(), input()))


    process_message(message_count)



notifications(int(input()))



'''
4
error
credit card purchase
Invalid customer address
500
warning
Email not confirmed
success
user registration
User registered successfully
warning
Customer has not email assigned
'''
